src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -0,0 +1,1325 @@
+        Get
+            Return 0
+        End Get
+    End Property
+    Public Property Length As Integer = 0
+End Class
+";
+            var csSource = @"
+class X
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+    } 
+}
+";
+            var vbCompilation = CreateVisualBasicCompilation(vbSource);
+            var csCompilation = CreateCompilation(csSource, parseOptions: TestOptions.RegularWithListPatterns, references: new[] { vbCompilation.EmitToImageReference() });
+            // PROTOTYPE(list-patterns) Unsupported because the lookup fails not that the indexer is static
+            csCompilation.VerifyEmitDiagnostics(
+                // (6,28): error CS9200: List patterns may not be used for a value of type 'Test1'.
+                //         _ = new Test1() is {0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{0}").WithArguments("Test1").WithLocation(6, 28));
+        }
+
+        [Theory]
+        [InlineData("public int this[Index i] { set {} }")]
+        [InlineData("public int this[int i, int ignored = 0] => 0;")]
+        [InlineData("public int this[params int[] i] => 0;")]
+        [InlineData("private int this[Index i] => 1;")]
+        public void ListPattern_MemberLookup_ErrorCases(string indexer)
+        {
+            var source = @"
+using System;
+class Test1
+{
+    " + indexer + @"
+    public int Length => 0;
+}
+class X
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+    } 
+}
+";
+            var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics(
+                // (12,28): error CS9200: List patterns may not be used for a value of type 'Test1'.
+                //         _ = new Test1() is {0};
+                Diagnostic(ErrorCode.ERR_UnsupportedTypeForListPattern, "{0}").WithArguments("Test1").WithLocation(12, 28));
+        }
+
+        [Fact]
+        public void ListPattern_MemberLookup_OverridenIndexer()
+        {
+            var source = @"
+using System;
+class Test1
+{
+    public virtual int this[Index i] => 1;
+    public int Count => 0;
+}
+class Test2 : Test1
+{
+}
+class Test3 : Test2
+{
+    public override int this[Index i] => 1;
+}
+class X
+{
+    public static void Main()
+    {
+        _ = new Test1() is {0};
+        _ = new Test2() is {0};
+        _ = new Test3() is {0};
+    } 
+}
+";
+            var compilation = CreateCompilationWithIndexAndRange(source, parseOptions: TestOptions.RegularWithListPatterns, options: TestOptions.ReleaseExe);
+            compilation.VerifyEmitDiagnostics();
+        }
+
+
+        [Fact]
+        public void ListPattern_RefReturns()
+        {
+            var source = @"
+using System;
+class Test1
+{
+    int value = 1;
+    public ref int this[Index i] => ref value;
```

Comment:
Nice scenario :-)
Consider also doing a `new Test1() is { var x }` and verifying the type of `var` and execution.